integration/exec_test.go

```diff
@@ -0,0 +1,209 @@
+		expectedMsgs []string
+	}{
+
+		{
+			description:  "fail due to action timeout",
+			action:       "action-fail-timeout",
+			shouldErr:    true,
+			expectedMsgs: []string{"context deadline exceeded"},
+		},
+		{
+			description:  "fail with fail fast",
+			action:       "action-fail-fast",
+			shouldErr:    true,
+			expectedMsgs: []string{`"task4" running container image "alpine:3.15.4" errored during run with status code: 1`},
+		},
+		{
+			description: "fail with fail safe",
+			action:      "action-fail-safe",
+			shouldErr:   true,
+			expectedMsgs: []string{
+				"[task5] hello-from-task5",
+				"[task5] bye-from-task5",
+				`* "task6" running container image "alpine:3.15.4" errored during run with status code: 1`,
+			},
+		},
+		{
+			description: "action succeeded",
+			action:      "action-succeeded",
+			envFile:     "exec.env",
+			expectedMsgs: []string{
+				"[task7] hello-from-env-file",
+				"[task7] bye-from-env-file",
+			},
+		},
+	}
+
+	for _, test := range tests {
+		testutil.Run(t, test.description, func(t *testutil.T) {
+			MarkIntegrationTest(t.T, CanRunWithoutGcp)
+			args := []string{test.action}
+
+			if test.envFile != "" {
+				args = append(args, "--env-file", test.envFile)
+			}
+
+			out, err := skaffold.Exec(args...).InDir("testdata/custom-actions-local").RunWithCombinedOutput(t.T)
+			t.CheckError(test.shouldErr, err)
+
+			for _, expectedMsg := range test.expectedMsgs {
+				t.CheckContains(expectedMsg, string(out))
+			}
+		})
+	}
+}
+
+func TestExec_LocalActionWithLocalArtifact(t *testing.T) {
+	tests := []struct {
+		description  string
+		action       string
+		shouldErr    bool
+		shouldBuild  bool
+		expectedMsgs []string
+	}{
+		{
+			description: "fail due not found image",
+			action:      "action-with-local-built-img",
+			shouldErr:   true,
+			expectedMsgs: []string{
+				"Error response from daemon: pull access denied for localtaks",
+			},
+		},
+		{
+			description: "build and run task",
+			action:      "action-with-local-built-img",
+			shouldBuild: true,
+			expectedMsgs: []string{
+				"[local-img-task1] Hello world from-local-img! 4",
+			},
+		},
+	}
+
+	for _, test := range tests {
+		testutil.Run(t, test.description, func(t *testutil.T) {
+			MarkIntegrationTest(t.T, CanRunWithoutGcp)
+			dir := "testdata/custom-actions-local"
+			args := []string{test.action}
+
+			if test.shouldBuild {
+				tmpfile := testutil.TempFile(t.T, "", []byte{})
+				skaffold.Build("--file-output", tmpfile).InDir(dir).RunOrFail(t.T)
+				args = append(args, "--build-artifacts", tmpfile)
+			}
+
+			out, err := skaffold.Exec(args...).InDir(dir).RunWithCombinedOutput(t.T)
+			t.CheckError(test.shouldErr, err)
+
+			for _, expectedMsg := range test.expectedMsgs {
+				t.CheckContains(expectedMsg, string(out))
+			}
+		})
+	}
+}
+
+func TestExec_ActionsEvents(t *testing.T) {
+	tests := []struct {
+		description  string
+		action       string
+		shouldErr    bool
+		expectedLogs []string
+	}{
+		// {
```

Comment:
Should this be commented out?

Reply:
No, it shouldn't be commented. I just changed it, thanks!